pgmax/groups/enumeration.py

```diff
@@ -0,0 +1,459 @@
+"""Defines EnumerationFactorGroup and PairwiseFactorGroup."""
+
+import collections
+from dataclasses import dataclass, field
+from typing import FrozenSet, Mapping, Optional, OrderedDict, Type, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from pgmax.factors import enumeration
+from pgmax.fg import groups, nodes
+
+
+@dataclass(frozen=True, eq=False)
+class EnumerationFactorGroup(groups.FactorGroup):
+    """Class to represent a group of EnumerationFactors.
+
+    All factors in the group are assumed to have the same set of valid configurations and
+    the same potential function. Note that the log potential function is assumed to be
+    uniform 0 unless the inheriting class includes a log_potentials argument.
+
+    Args:
+        factor_configs: Array of shape (num_val_configs, num_variables)
+            An array containing explicit enumeration of all valid configurations
+    """
+
+    factor_configs: np.ndarray = field(init=True)
+    log_potentials: np.ndarray = field(init=True, default=np.empty((0,)))
```

Comment:
Get rid of `field`. Also why do we need to change the original `Optional[np.ndarray] = None` here? `None` makes more sense than empty in this case

Reply:
the parent class will assing log_potentials to `np.empty((0,))` by default, so adding the None option is adding noise